static/templates/recent_topic_row.hbs

```diff
@@ -1,18 +1,43 @@
-<tr id="recent_topic:{{topic_key}}" {{#if unread_count}}class="unread_topic"{{/if}} data-unread-count="{{unread_count}}" data-muted="{{muted}}" data-participated="{{participated}}">
+<tr id="recent_conversation:{{conversation_key}}" {{#if unread_count}}class="unread_topic"{{/if}} data-unread-count="{{unread_count}}" data-muted="{{muted}}" data-participated="{{participated}}">
     <td class="recent_topic_stream">
+        {{#if is_private}}
+        <div class="flex_container flex_container_pm">
+            <div class="left_part recent_topics_focusable">
+                <span class="fa fa-envelope"></span>
+                <a href="{{pm_url}}">Private messages</a>
+            </div>
+            <div class="right_part">
+                <span class="stream-privacy filter-icon">
+                    {{#if is_group}}
+                    <span class="fa fa-group"></span>
+                    {{else}}
+                    <span class="{{user_circle_class}} user_circle"></span>
+                    {{/if}}
+                </span>
+            </div>
+        </div>
+        {{else}}
         <div class="recent_topics_focusable">
             <span id="stream_sidebar_privacy_swatch_{{stream_id}}" class="stream-privacy filter-icon" style="color: {{stream_color}}">
                 {{> stream_privacy }}
             </span>
-            <a href="{{stream_url}}">{{stream}}</a>
+            <a href="{{topic_url}}">{{stream}}</a>
         </div>
+        {{/if}}
     </td>
     <td class="recent_topic_name">
         <div class="flex_container">
             <div class="left_part recent_topics_focusable">
+                {{#if is_private}}
+                <a href="{{pm_url}}">{{pm_with}}</a>
+                {{else}}
                 <a href="{{topic_url}}">{{topic}}</a>
+                {{/if}}
             </div>
             <div class="right_part">
+                {{#if is_private}}
+                <span class="unread_count unread_count_pm {{#unless unread_count}}unread_hidden{{/unless}}">{{unread_count}}</span>
```

Comment:
Why is this if/else outside the `recent_topics_actions` block? Oh, I guess we have this existing weird setup -- the name of `recent_topics_actions` suggests there is a single div, but there's actually just one.

And I guess it's intentional that you can't mark as read from this UI, but that seems like something we'd want to change in a follow-up?

Reply:
Mark as read is possible from this UI, it just takes some time to rerender and remove the unread counter.

Reply:
No, if you hover over a private message count, one can't mark it as read here. I think that's probably a bug?